### PR TITLE
removes a bunch of unused/bad helpers

### PR DESCRIPTION
**Pull request by vuonojenmustaturska**
I wouldn't mind leaving `safepick()` if it was at least used responsibly, but more than half the time the code would just runtime from it returning a null or the list is guaranteed to be not empty.

I also yeeted some end-of-proc empty returns from mechacode because I was there.

### CODE REVIEW

**Comment by spookydonut** [comment]
https://github.com/spookydonut/SpacemanDMM/tree/deadcode

File: code/modules/jobs/job_types/security_officer.dm [hunk 1]
```diff
@@ -94,14 +94,20 @@ GLOBAL_LIST_INIT(available_depts, list(SEC_DEPT_ENGINEERING, SEC_DEPT_MEDICAL, S
...
-				else
-					break
+			var/list/possible_turfs = get_area_turfs(destination)
+			while (length(possible_turfs))
+				var/I = rand(1, possible_turfs.len)
+				var/turf/target = possible_turfs[I]
+				if(!target.density)
+					var/clear = TRUE
+					for(var/obj/O in target)
+						if(O.density)
```
Comment (ninjanomnom) [inline code/modules/jobs/job_types/security_officer.dm]: Use CanPass instead of writing your own

**Review by ninjanomnom — CHANGES_REQUESTED** [review]

**Review by spookydonut — APPROVED** [review]
Naksu asked me to review, looks good and procs removed are deadcode

**Comment by vuonojenmustaturska** [comment]
i thought this guy was a maintainer

File: code/__HELPERS/text.dm [hunk 2]
```diff
@@ -656,7 +656,7 @@ GLOBAL_LIST_INIT(binary, list("0","1"))
 	if(fexists(log))
 		oldjson = json_decode(file2text(log))
 		oldentries = oldjson["data"]
-	if(!isemptylist(oldentries))
+	if(!!length(oldentries))
```
Comment (AnturK) [inline code/__HELPERS/text.dm]: !!

File: code/__HELPERS/unsorted.dm [hunk 3]
```diff
@@ -1106,7 +1106,9 @@ B --><-- A
 	A.cut_overlay(O)
 
 /proc/get_random_station_turf()
-	return safepick(get_area_turfs(pick(GLOB.the_station_areas)))
+	var/list/turfs = get_area_turfs(pick(GLOB.the_station_areas))
+	if (length(turfs))
+		return pick(get_area_turfs(pick(GLOB.the_station_areas)))
```
Comment (AnturK) [inline code/__HELPERS/unsorted.dm]: just pick(turfs) here ?

**Review by AnturK — CHANGES_REQUESTED** [review]

File: code/__HELPERS/unsorted.dm (hunk 3 above)
Comment (spookydonut) [reply code/__HELPERS/unsorted.dm]: yeah pick() is fine with empty lists you just get null

Comment (vuonojenmustaturska) [reply code/__HELPERS/unsorted.dm]: pick() from an empty list is a runtime

**Pull request merged**